src/algebra/quaternion.lean

```diff
@@ -209,6 +223,37 @@ variables (R c₁ c₂)
 @[simps] def im_k_lm : ℍ[R, c₁, c₂] →ₗ[R] R :=
 { to_fun := im_k, map_add' := λ x y, rfl, map_smul' := λ r x, rfl }
 
+/-- `quaternion_algebra.equiv_tuple` as a linear equivalence. -/
+def linear_equiv_tuple : ℍ[R,c₁,c₂] ≃ₗ[R] (fin 4 → R) :=
+linear_equiv.symm
+  { to_fun := (equiv_tuple c₁ c₂).symm,
+    inv_fun := (equiv_tuple c₁ c₂),
+    map_add' := λ v₁ v₂, rfl,
+    map_smul' := λ v₁ v₂, rfl,
+    .. (equiv_tuple c₁ c₂).symm }
```

Comment:
Sorry if I'm missing something obvious, but why do you need to `symm` everything?
```suggestion
def linear_equiv_tuple : ℍ[R,c₁,c₂] ≃ₗ[R] (fin 4 → R) :=
{ map_add' := λ v₁ v₂, rfl,
  map_smul' := λ v₁ v₂, rfl,
  ..equiv_tuple c₁ c₂ }
```

Reply:
I seem to remember that that didn't work as the proofs weren't rfl in that direction; but I can't test right now because the cache seems to be a corrupt tarball...

Reply:
Yep, I can confirm that `rfl` doesn't work in your code. I added a comment explaining the `symm`.
You could of course spend some effor proving the field, but I don't see any reason to do so.